### PR TITLE
Add `state` & `variables` to GetExhangeResponse.

### PR DESCRIPTION
**Pull request by BigBlueHat**
Can't track `state` if it's not there. 😃

### CODE REVIEW

File: exchanges.yml [hunk 1]
```diff
@@ -386,20 +386,39 @@ components:
...
-          description: The steps completed by the exchange previously.
-          items:
-            type: string
+        exchange:
+          type: object
+          description: The active exchange.
+          properties:
+            exchangeId:
+              type: string
+              description: The URL that uniquely identifies the exchange.
```
Comment (dlongley) [inline exchanges.yml:395]:
```suggestion
            id:
              type: string
              description: The local exchange ID that identifies the exchange.
```

File: exchanges.yml [hunk 2]
```diff
@@ -386,20 +386,39 @@ components:
...
+              type: string
+              description: The time to live for the exchange.
+            nextStep:
+              type: string
+              description: The interaction the exchange is expecting to happen next.
+            completedSteps:
+              type: array
+              description: The steps completed by the exchange previously.
+              items:
+                type: string
```
Comment (dlongley) [inline exchanges.yml:406]: Simplified to match what's in the exchange.yml:

```suggestion
            step:
              type: string
              description: The current step in the exchange.
```

File: exchanges.yml [hunk 3]
```diff
@@ -386,20 +386,39 @@ components:
...
+              description: Template variables to be used in the exchange.
+              items:
+                type: object
+                properties:
+                  type:
+                    type: string
+                    description: The type of template.
+                  template:
+                    type: string
+                    description: The template itself.
```
Comment (dlongley) [inline exchanges.yml:421]: This is actually a map. Probably needs fixing in the exchange.yml as well.

```suggestion
            variables:
              type: object
              description: Template variables to be used in the exchange.
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: exchanges.yml (hunk 3 above)
Comment (BigBlueHat) [reply exchanges.yml]: I think you mean in `CreateExchangeResponse`? There is no `exchange.yml` that I could find.

**Comment by BigBlueHat** [comment]
@dlongley let me know if you want more cleanup throughout. I'd prefer we merge this (if it's ready), though, and make an issue (or another PR) for further cleanup.

**Review by dlongley — APPROVED** [review]

**Review by msporny — APPROVED** [review]

File: exchanges.yml [hunk 4]
```diff
@@ -386,20 +386,25 @@ components:
...
+              description: The local exchange ID that identifies the exchange.
+            ttl:
+              type: string
+              description: The time to live for the exchange.
+            step:
+              type: string
+              description: The current step in the exchange.
+            state:
+              type: string
+              description: The status ("pending" | "complete" | "invalid") of the exchange, set to "pending" on creation.
```
Comment (msporny) [inline exchanges.yml:404]:
```suggestion
              description: The status ("pending" | "active" | "complete" | "invalid") of the exchange, set to "pending" on creation.
```

**Comment by TallTed** [comment]
Even though already merged, it seems worth correcting the spelling of `GetExhangeResponse` in the title (and initial commit) of this PR, to `GetExchangeResponse`.

**Pull request merged**